### PR TITLE
Feature/esm jest

### PR DESCRIPTION
**Pull request by niemyjski**
Upgraded tests to jest and dropped cjs for esm. Everyone is converting their libraries to esm which supports importing cjs but not the other way around. I was running into issues importing and consuming parse from esm due to a `self` reference.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "presets": [
+        "@babel/preset-env"
```
Comment (niemyjski) [inline .babelrc:3]: This is currently required for jest but I expect soon we can drop babel for tests.

File: __tests__/get-test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { get } from "../index.js";
+
+describe("get", () => {
+  test("basic", () => {
+    (function testBasic() {
+      var trace = get();
+
+      //expect(trace[0].getFunction()).toBe(testBasic);
```
Comment (niemyjski) [inline __tests__/get-test.js:8]: @felixge I'm not sure why but getFunction always returns undefined/null here

File: __tests__/get-test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      expect(trace[0].getFunctionName()).toBe('testBasic');
+      expect(trace[0].getFileName()).toBe(__filename);
+    })();
+  });
+
+  test("wrapper", () => {
+    (function testWrapper() {
+      (function testBelowFn() {
+        var trace = get(testBelowFn);
+        //expect(trace[0].getFunction()).toBe(testWrapper);
```
Comment (niemyjski) [inline __tests__/get-test.js:18]: @felixge I'm not sure why but getFunction always returns undefined/null here

File: __tests__/parse-test.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        //const realJson = JSON.stringify(realValue);
+        //const parsedJson = JSON.stringify(parsedValue);
+        //console.log(method + ': ' + realJson + ' != ' + parsedJson + ' (#' + i + ')');
+        expect(realValue).toBe(parsedValue);
+      }
+
+      compare('getFileName');
+      compare('getFunctionName', {
+        2: 'Object.asyncJestTest',
+        4: 'new Promise'
```
Comment (niemyjski) [inline __tests__/parse-test.js:115]: This feels weird, but we are parsing what is passed to us. It probably could be better

File: __tests__/parse-test.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        2: 'asyncJestTest'
+      });
+      compare('getLineNumber', {
+        0: 88,
+        1: 92
+      });
+      compare('getColumnNumber', {
+        0: 13
+      });
+      compare('isNative');
```
Comment (niemyjski) [inline __tests__/parse-test.js:130]: These overrides seem brittle depending on node version?

File: index.js [hunk 6]
```diff
@@ -1,34 +1,32 @@
...
   }
 
-  var self = this;
-  var lines = err.stack.split('\n').slice(1);
-
+  const lines = err.stack.split('\n').slice(1);
   return lines
     .map(function(line) {
       if (line.match(/^\s*[-]{4,}$/)) {
-        return self._createParsedCallSite({
```
Comment (niemyjski) [inline index.js:31]: This is the whole reason I migrated to esm. When this script is used in esm (you can import cjs) this self reference was globalThis and caused runtime issues.

**Review by niemyjski — COMMENTED** [review]
@felixge Can you let me know your thoughts? It would also be nice to upgrade to github actions at some point but I do not have time for that atm.

File: __tests__/get-test.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import { get } from "../index.js";
```
Comment (niemyjski) [inline __tests__/get-test.js:1]: I moved and renamed these tests, not sure why it thinks it was a delete and new instead of a `git mv`

**Comment by niemyjski** [comment]
@felixge I'd like your thoughts before merging this unless you are fine with these changes since they are breaking (dropping cjs). I'm currently blocked on my esm project.

**Comment by felixge** [comment]
@niemyjski this LGTM. I don't have time for an in-depth review. My only thought is that perhaps it's time to give this a major (v1) version number to make it clear it's a breaking changing and then keep doing the semver thing going forward.

**Pull request merged**